[Bug]: AgentBuilder doesn't build agent based on the specified agent_path

### Describe the bug

`AgentBuilder` doesn't use `agent_path` if it is specified in the `agent_configs`, it just creates a default `AssistantAgent`

### Steps to reproduce

```python
from autogen.agentchat.contrib.captainagent.agent_builder import AgentBuilder
from autogen.agents.experimental import WebSurferAgent


agent_builder = AgentBuilder(
    # config_file_or_env=OAI_CONFIG_LIST,
    # config_file_location=KEY_LOC,
    builder_model_tags=["gpt-4o"],
    agent_model_tags=["gpt-4o"],
)

conf = {
    "building_task": "Generate one WebSurferAgent for scraping wikipedia",
    "agent_configs": [
        {
            "name": "WebSurferAgent",
            "model": ["gpt-4o"],
            "description": "A helpful assistant with access to a web browser. Ask them to perform web searches, open pages, navigate to Wikipedia, answer questions from pages, and or generate summaries.",
            "system_message": "",
            "agent_path": "autogen/agents/experimental/websurfer/websurfer/WebSurferAgent",
            "web_tool_kwargs": {
                "agent_kwargs": {
                    "max_steps": 100
                }
            }
        }
    ],
    "coding": True,
    "default_llm_config": {
        "temperature": 0
    },
    "code_execution_config": {
        "work_dir": ".",
        "use_docker": False,
        "timeout": 60,
        "last_n_messages": 2
    },
}

agents, _ = agent_builder.build(**conf)

websurfer_in_agents = False
for agent in agents:
    if isinstance(agent, WebSurferAgent):
        websurfer_in_agents = True
        break
    
assert websurfer_in_agents, "WebSurferAgent not found in agents"

```

```console
==> Generating agents...
['WebScraping_Expert', 'Wikipedia_Expert', 'Python_Expert', 'Shell_Expert', 'DataExtraction_Expert'] are generated.
==> Generating system message...
Preparing system message for WebScraping_Expert
Preparing system message for Wikipedia_Expert
Preparing system message for Python_Expert
Preparing system message for Shell_Expert
Preparing system message for DataExtraction_Expert
==> Generating description...
Preparing description for WebScraping_Expert
Preparing description for Wikipedia_Expert
Preparing description for Python_Expert
Preparing description for Shell_Expert
Preparing description for DataExtraction_Expert
==> Creating agents...
Creating agent WebScraping_Expert...
Creating agent Wikipedia_Expert...
Creating agent Python_Expert...
Creating agent Shell_Expert...
Creating agent DataExtraction_Expert...
Adding user console proxy...
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
Cell In[2], [line 47](vscode-notebook-cell:?execution_count=2&line=47)
     [45](vscode-notebook-cell:?execution_count=2&line=45)         websurfer_in_agents = True
     [46](vscode-notebook-cell:?execution_count=2&line=46)         break
---> [47](vscode-notebook-cell:?execution_count=2&line=47) assert websurfer_in_agents, "WebSurferAgent not found in agents"

AssertionError: WebSurferAgent not found in agents
```

### Model Used

_No response_

### Expected Behavior

_No response_

### Screenshots and logs

_No response_

### Additional Information

_No response_

--- a/autogen/agentchat/contrib/captainagent/agent_builder.py
+++ b/autogen/agentchat/contrib/captainagent/agent_builder.py
@@ -397,7 +397,7 @@ Match roles in the role set to each expert in expert set.
         if max_agents is None:
             max_agents = self.max_agents
 
-        agent_configs = [] if "agent_configs" not in kwargs else kwargs["agent_configs"]
+        agent_configs = kwargs.get("agent_configs", [])
         self.building_task = building_task
 
         print(colored("==> Generating agents...", "green"), flush=True)

--- a/autogen/agentchat/contrib/captainagent/agent_builder.py
+++ b/autogen/agentchat/contrib/captainagent/agent_builder.py
@@ -397,7 +397,7 @@ Match roles in the role set to each expert in expert set.
         if max_agents is None:
             max_agents = self.max_agents
 
-        agent_configs = []
+        agent_configs = [] if "agent_configs" not in kwargs else kwargs["agent_configs"]
         self.building_task = building_task
 
         print(colored("==> Generating agents...", "green"), flush=True)

--- a/test/agentchat/contrib/test_agent_builder.py
+++ b/test/agentchat/contrib/test_agent_builder.py
@@ -13,7 +13,7 @@ import tempfile
 import pytest
 
 from autogen.agentchat.contrib.captainagent.agent_builder import AgentBuilder
-from autogen.agents.experimental.websurfer.websurfer import WebSurferAgent
+from autogen.agentchat.contrib.text_analyzer_agent import TextAnalyzerAgent
 from autogen.import_utils import optional_import_block, skip_on_missing_imports
 
 from ...conftest import KEY_LOC, OAI_CONFIG_LIST
@@ -125,19 +125,16 @@ def test_build_from_library(builder: AgentBuilder):
 
 @pytest.mark.openai
 @skip_on_missing_imports(["openai"], "openai")
-@skip_on_missing_imports(["langchain_openai", "browser_use"], "browser-use")
 def test_build_with_agent_configs(builder: AgentBuilder):
     conf = {
-        "building_task": "Generate one WebSurferAgent for scraping wikipedia",
+        "building_task": "Generate one TextAnalyzerAgent to analyze text",
         "agent_configs": [
             {
-                "name": "WebSurferAgent",
+                "name": "TextAnalyzerAgent",
                 "model": ["gpt-4o"],
-                "description": "A helpful assistant with access to a web browser. Ask them to perform web searches, open pages, navigate to Wikipedia, answer questions from pages, and or generate summaries.",
+                "description": "A helpful assistant to analyze text. Ask them to analyze any text, and they will provide you with the analysis.",
                 "system_message": "",
-                "agent_path": "autogen/agents/experimental/websurfer/websurfer/WebSurferAgent",
-                # "agent_path": "autogen.agents.experimental.WebSurferAgent",
-                "web_tool_kwargs": {"agent_kwargs": {"max_steps": 100}},
+                "agent_path": "autogen/agentchat/contrib/text_analyzer_agent/TextAnalyzerAgent",
             }
         ],
         "coding": True,
@@ -147,13 +144,13 @@ def test_build_with_agent_configs(builder: AgentBuilder):
 
     agents, _ = builder.build(**conf)
 
-    websurfer_in_agents = False
+    is_agent_found = False
     for agent in agents:
-        if isinstance(agent, WebSurferAgent):
-            websurfer_in_agents = True
+        if isinstance(agent, TextAnalyzerAgent):
+            is_agent_found = True
             break
 
-    assert websurfer_in_agents, "WebSurferAgent not found in agents"
+    assert is_agent_found, "TextAnalyzerAgent not found in agents"
 
 
 @pytest.mark.openai

--- a/test/agentchat/contrib/test_agent_builder.py
+++ b/test/agentchat/contrib/test_agent_builder.py
@@ -13,6 +13,7 @@ import tempfile
 import pytest
 
 from autogen.agentchat.contrib.captainagent.agent_builder import AgentBuilder
+from autogen.agents.experimental.websurfer.websurfer import WebSurferAgent
 from autogen.import_utils import optional_import_block, skip_on_missing_imports
 
 from ...conftest import KEY_LOC, OAI_CONFIG_LIST
@@ -120,6 +121,39 @@ def test_build_from_library(builder: AgentBuilder):
 
     # check number of agents
     assert len(agent_config["agent_configs"]) <= builder.max_agents
+
+
+@pytest.mark.openai
+@skip_on_missing_imports(["openai"], "openai")
+@skip_on_missing_imports(["langchain_openai", "browser_use"], "browser-use")
+def test_build_with_agent_configs(builder: AgentBuilder):
+    conf = {
+        "building_task": "Generate one WebSurferAgent for scraping wikipedia",
+        "agent_configs": [
+            {
+                "name": "WebSurferAgent",
+                "model": ["gpt-4o"],
+                "description": "A helpful assistant with access to a web browser. Ask them to perform web searches, open pages, navigate to Wikipedia, answer questions from pages, and or generate summaries.",
+                "system_message": "",
+                "agent_path": "autogen/agents/experimental/websurfer/websurfer/WebSurferAgent",
+                # "agent_path": "autogen.agents.experimental.WebSurferAgent",
+                "web_tool_kwargs": {"agent_kwargs": {"max_steps": 100}},
+            }
+        ],
+        "coding": True,
+        "default_llm_config": {"temperature": 0},
+        "code_execution_config": {"work_dir": ".", "use_docker": False, "timeout": 60, "last_n_messages": 2},
+    }
+
+    agents, _ = builder.build(**conf)
+
+    websurfer_in_agents = False
+    for agent in agents:
+        if isinstance(agent, WebSurferAgent):
+            websurfer_in_agents = True
+            break
+
+    assert websurfer_in_agents, "WebSurferAgent not found in agents"
 
 
 @pytest.mark.openai

